### PR TITLE
Doker file for NMMA

### PR DESCRIPTION
**Pull request by FabioRagosta**
[nmma_docker.zip](https://github.com/nuclear-multimessenger-astronomy/nmma/files/14277480/nmma_docker.zip)

This docker file allows to create an image of Ubunti20.04 Operative System and to install nmma and all the needed dependeces.

### CODE REVIEW

**Comment by sahiljhawar** [comment]
@FabioRagosta Any specific reason to change minimum `astropy` version?

**Comment by sahiljhawar** [comment]
@FabioRagosta Along with this can you provide CI/CD pipeline to publish the image on dockerhub? If you need help you can see the example Docker CI here: [build-deploy-container.yml](https://github.com/nuclear-multimessenger-astronomy/nmma/blob/main/.github/workflows/build-deploy-container.yml)

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    git \
+    python3-pip \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+
+# Set work directoryI
+WORKDIR /work
+
+# Clone a Git repository using the provided credentials
+RUN git clone https://github.com/FabioRagosta/nmma/
```
Comment (bfhealy) [inline Dockerfile:19]: Should this line instead clone `https://github.com/nuclear-multimessenger-astronomy/nmma/`?

File: requirements.txt [hunk 2]
```diff
@@ -6,7 +6,7 @@ numpy>=1.9
 matplotlib>=2.0,<3.8
 scipy>=1.10
 pandas>=1.3.4,<2.0
-astropy>=5.3.3
+astropy>=4.3.1
```
Comment (bfhealy) [inline requirements.txt:9]: I echo Sahil's question about the astropy version here.

**Review by bfhealy — COMMENTED** [review]
Hi @FabioRagosta, thanks for opening this PR! Just a few questions on the details.

File: Dockerfile (hunk 1 above)
Comment (FabioRagosta) [reply Dockerfile:19]: yes it should, I forgot to change it before uploading. I used the forked repository for the trial.

File: requirements.txt (hunk 2 above)
Comment (FabioRagosta) [reply requirements.txt:9]: The reason is just related to the installation through docker. While I was testing I often got an error that pip could not found any compatible version. However changing the minimum version no error was raised again.

**Comment by FabioRagosta** [comment]
> @FabioRagosta Along with this can you provide CI/CD pipeline to publish the image on dockerhub? If you need help you can see the example Docker CI here: [build-deploy-container.yml](https://github.com/nuclear-multimessenger-astronomy/nmma/blob/main/.github/workflows/build-deploy-container.yml)
I added the CI/CD pipeline in this zip. 
[nmma_docker.zip](https://github.com/nuclear-multimessenger-astronomy/nmma/files/14297670/nmma_docker.zip)

**Comment by sahiljhawar** [comment]
@FabioRagosta add it to the workflows folder through a commit. Zip file is of no use for end user.

**Review by bfhealy — APPROVED** [review]
LGTM. @sahiljhawar, do you think the .yml file should be located with the others in the `.github/workflows` directory, or is it good where it is?

**Comment by sahiljhawar** [comment]
@bfhealy I think it should be in the workflows directory. Only then the image will be pushed to Dockerhub

File: build-deploy-container_nmma.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        id: buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKER_HUB_USERNAME }}
+          password: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
+      
+    - name: Build Docker image
```
Comment (sahiljhawar) [inline build-deploy-container_nmma.yml]: If you open this file in Github editor the following error pops up:
`Bad indentation of a mapping entry`

**Comment by sahiljhawar** [comment]
@bfhealy I am not sure how Docker CI really works. I think only way to test this CI is to merge this and see if it really works or not

**Review by sahiljhawar — APPROVED** [review]

**Pull request merged**